### PR TITLE
several Updates:

### PR DESCRIPTION
**Pull request by tuxmea**
- Update Ruby Version
- Update Puppet Modules
- Updates for onceover
- Update PE Version for Vagrant
- Add r10k to Gemfile

### CODE REVIEW

File: Puppetfile [hunk 1]
```diff
@@ -9,6 +9,22 @@ mod 'example42/hieradata',
   :branch => :control_branch,
   :default_branch => 'production'
 
+# Puppet 6 Core Modues
```
Comment (alvagante) [inline Puppetfile]: What happens if one uses this Puppetfile with Puppet pre 6?

**Comment by tuxmea** [comment]
> Wow, looks great. Did you test it with onceover?

No. This is just preparation.

File: Puppetfile (hunk 1 above)
Comment (tuxmea) [reply Puppetfile]: Good question. I wrote to puppet-users for feedback.

Comment (tuxmea) [reply Puppetfile]: https://groups.google.com/forum/#!msg/puppet-users/HMgdFb-cVy8/_JjyHSy7CAAJ
Puppet ensures compatibility.

**Pull request merged**